[Gstreamer][HLS] Unable to play TED videos

philn opened the pull request:
#### e2afda4dd0ab875371bd993ec2a1a8495a142026
<pre>
[Gstreamer][HLS] Unable to play TED videos
<a href="https://bugs.webkit.org/show_bug.cgi?id=174458">https://bugs.webkit.org/show_bug.cgi?id=174458</a>

Reviewed by Xabier Rodriguez-Calvar.

Disable HLS support by default in GStreamer ports. The underlying GStreamer hlsdemux element is not
receiving much maintenance and its replacement, hlsdemux2, cannot be used by WebKit due to
sandboxing requirements.

Most websites should now fallback to MSE if HLS is not supported by the User-Agent. This was checked
on TED and Apple developer websites.

HLS support can be re-enabled at runtime by setting the `WEBKIT_GST_ENABLE_HLS_SUPPORT=1`
environment variable.

* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::initializeDecoders):

Canonical link: <a href="https://commits.webkit.org/258717@main">https://commits.webkit.org/258717@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c231120e237df5fe2eddd6784a95db4008ef30de

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/102786 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/11906 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/35813 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/112044 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/172267 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/106752 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/12919 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/2815 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/95038 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/109721 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/108561 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/9916 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/93123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/37562 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/91767 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/24643 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/79301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/5361 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26060 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/5510 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/2511 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/11529 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/45553 "Passed tests") | | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5988 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/7253 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

calvaris commented on Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
```diff
@@ -416,6 +415,12 @@ void GStreamerRegistryScanner::initializeDecoders(const GStreamerRegistryScanner
     };
+
+    if (const char* hlsSupport = getenv("WEBKIT_GST_ENABLE_HLS_SUPPORT")) {
+        if (!strcmp(hlsSupport, "1"))
```
Please, use `g_getenv` and `g_strcmp0` instead.

philn replied:
Sorry I don't remember about `g_getenv`, is this some policy? Because we have a mix  of `getenv` and `g_getenv` currently. About `g_strcmp0` well we're already guaranteed here that hlsSupport is not NULL, so I'm not sure there's a point :)

calvaris replied:
I think both are safer and more platform compatible, specially if code changes and for some reason `strcmp` is not protected by the if above.

philn committed:
[Gstreamer][HLS] Unable to play TED videos
https://bugs.webkit.org/show_bug.cgi?id=174458

Reviewed by Xabier Rodriguez-Calvar.

Disable HLS support by default in GStreamer ports. The underlying GStreamer hlsdemux element is not
receiving much maintenance and its replacement, hlsdemux2, cannot be used by WebKit due to
sandboxing requirements.

Most websites should now fallback to MSE if HLS is not supported by the User-Agent. This was checked
on TED and Apple developer websites.

HLS support can be re-enabled at runtime by setting the `WEBKIT_GST_ENABLE_HLS_SUPPORT=1`
environment variable.

* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::initializeDecoders):

Canonical link: https://commits.webkit.org/258717@main
